pyartcd/pyartcd/pipelines/build_microshift.py

```diff
@@ -139,34 +139,68 @@ def _generate_example_schema(self, nvrs: List[str]) -> Dict:
-                                {
-                                    "distgit_key": dg_key,
-                                    "metadata": {
-                                        "is": is_entry
-                                    }
-                                }
-                            ]
-                        }
-                    }
-                }
-            }
-        }
-        return schema
+
+        rpms_entry = releases_config["releases"][self.assembly].setdefault("assembly", {}).setdefault("members", {}).setdefault("rpms", [])
+        microshift_entry = next(filter(lambda rpm: rpm.get("distgit_key") == dg_key, rpms_entry), None)
+        if microshift_entry is None:
+            microshift_entry = {"distgit_key": dg_key}
+            rpms_entry.append(microshift_entry)
+        microshift_entry.setdefault("metadata", {})["is"] = is_entry
+        return microshift_entry
+
+    async def _create_or_update_pull_request(self, nvrs: List[str]):
+        self._logger.info("Creating ocp-build-data PR...")
+        # Clone ocp-build-data
+        build_data_path = self._working_dir / "ocp-build-data-push"
+        build_data = GitRepository(build_data_path, dry_run=self.runtime.dry_run)
+        ocp_build_data_repo_push_url = self.runtime.config["build_config"]["ocp_build_data_repo_push_url"]
+        await build_data.setup(ocp_build_data_repo_push_url)
+        branch = f"auto-pin-microshift-{self.group}-{self.assembly}"
+        await build_data.fetch_switch_branch(branch, self.group)
+        # Make changes
+        releases_yaml_path = build_data_path / "releases.yml"
+        releases_yaml = yaml.load(releases_yaml_path)
+        self._pin_nvrs(nvrs, releases_yaml)
+        yaml.dump(releases_yaml, releases_yaml_path)
+        # Create a PR
+        title = f"Pin microshift build for {self.group} {self.assembly}"
+        body = f"Created by job run {self.runtime.get_job_run_url()}"
+        match = re.search(r"github\.com[:/](.+)/(.+)(?:.git)?", ocp_build_data_repo_push_url)
+        if not match:
+            raise ValueError(f"Couldn't create a pull request: {ocp_build_data_repo_push_url} is not a valid github repo")
+        head = f"{match[1]}:{branch}"
+        base = self.group
+        if self.runtime.dry_run:
+            self._logger.warning("[DRY RUN] Would have created pull-request with head '%s', base '%s' title '%s', body '%s'", head, base, title, body)
+            d = {"html_url": "https://github.example.com/foo/bar/pull/1234", "number": 1234}
+            result = namedtuple('pull_request', d.keys())(*d.values())
+            return result
+        pushed = await build_data.commit_push(f"{title}\n{body}")
+        result = None
+        if pushed:
+            github_token = os.environ.get('GITHUB_TOKEN')
+            if not github_token:
+                raise ValueError("GITHUB_TOKEN environment variable is required to create a pull request")
+            owner = "openshift"
```

Comment:
Could the branch be pushed to the bot's account, and a PR opened from there?

Reply:
he repo to push is actually configured at https://github.com/openshift/aos-cd-jobs/blob/94905e9c5c1a7e381cdda8b9a85cb0fea7f5254e/config/artcd.toml#L8. It can be changed to the fork, but we need some tests (and maybe also need to pre-create the fork) before we actually make that change.

The `owner` here is the merge destination.